### PR TITLE
Fix React warnings

### PR DESCRIPTION
**Pull request by somebody1234**
### Pull Request Description
Fix some React warnings:
- Fix warning about nested `button`s when close button is in tab button
  - The fix for now is to change the outer `button` to a link without a `href`. It's not ideal but we still want to use the CSS that comes with buttons, we just can't have nested `button` elements in HTML.
- Fix warning about setting state outside of a `useEffect` when deciding to show Local category when user is not enabled
- Fix passing invalid `isInvalid` prop to plain HTML element in ToS modal

### Important Notes
None

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] ~~The documentation has been updated, if necessary.~~
- [x] ~~Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.~~
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      [TypeScript](https://github.com/enso-org/enso/blob/develop/docs/style-guide/typescript.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- [x] ~~Unit tests have been written where possible.~~

### CODE REVIEW

File: app/ide-desktop/lib/dashboard/src/layouts/TabBar.tsx [hunk 1]
```diff
@@ -218,6 +218,8 @@ export function Tab(props: InternalTabProps) {
       )}
     >
       <ariaComponents.Button
+        // Change this `button` into an `a` to avoid the DOM nesting error.
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/layouts/TabBar.tsx]: This is not acceptable by the aria guidelines. We can't put anything interactive inside a link.

Comment (MrFlashAccount) [reply app/ide-desktop/lib/dashboard/src/layouts/TabBar.tsx]: I think the best solution is to use  https://react-spectrum.adobe.com/react-aria/Tabs.html

**Comment by somebody1234** [comment]
@MrFlashAccount here's the wip btw - based on the debug logs it seems like the `clip-path` *is* getting set, and then it is immediately getting unset by a *second* ref containing `null`, in a component that somehow has a reference to the *same* background element

on a side note, i've removed `loaderPosition` from `ariaComponents.Button` since i imagine it was only useful for tab buttons - the default behavior is probably what's best for all actual buttons

**Comment by MrFlashAccount** [comment]
> on a side note, i've removed loaderPosition from ariaComponents.Button since i imagine it was only useful for tab buttons - the default behavior is probably what's best for all actual buttons

This one is a good thing. I think we should probably keep it

**Comment by somebody1234** [comment]
@MrFlashAccount ready for review i guess.

i ended up using a different solution to yours, as the changes made to the markup resulted in the user bar (user menu, share buttons etc) taking up the entire vertical space

File: app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx [hunk 2]
```diff
@@ -64,9 +64,6 @@ export interface BaseButtonProps<Render>
   readonly testId?: string
   readonly isDisabled?: boolean
   readonly formnovalidate?: boolean
-  /** Defaults to `full`. When `full`, the entire button will be replaced with the loader.
-   * When `icon`, only the icon will be replaced with the loader. */
-  readonly loaderPosition?: 'full' | 'icon'
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx]: Can we return this functionality please?

File: app/ide-desktop/lib/dashboard/src/pages/dashboard/Dashboard.tsx [hunk 3]
```diff
@@ -234,13 +237,15 @@ function DashboardInner(props: DashboardProps) {
...
-  const isUserEnabled = user.isEnabled
-
-  const defaultCategory = initialLocalProjectId == null ? Category.cloud : Category.local
 
   const [category, setCategory] = searchParamsState.useSearchParamsState(
     'driveCategory',
-    () => defaultCategory,
+    () => {
+      const isUserEnabled = user.isEnabled
+      const canUserAccessCloud = isUserEnabled || localBackend == null
```
Comment (MrFlashAccount) [inline app/ide-desktop/lib/dashboard/src/pages/dashboard/Dashboard.tsx]: `canUserAccessCloud` is misleading here, user *can* access cloud if localBackend != null

**Review by MrFlashAccount — APPROVED** [review]
LGTM aside a few comments down below

File: app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx (hunk 2 above)
Comment (somebody1234) [reply app/ide-desktop/lib/dashboard/src/components/AriaComponents/Button/Button.tsx]: oh i thought i had returned it, i guess it got un-returned by the force pushes

**Pull request merged**